addons/l10n_in/models/account_invoice.py

```diff
@@ -67,6 +68,22 @@ def _compute_l10n_in_state_id(self):
             else:
                 move.l10n_in_state_id = False
 
+    @api.depends('partner_id', 'partner_shipping_id', 'company_id', 'l10n_in_state_id')
+    def _compute_fiscal_position_id(self):
+        in_moves = self.filtered(lambda move: move.country_code == 'IN' and move.journal_id.type == 'sale')
+        for move in in_moves:
+            country_id = move.l10n_in_state_id.country_id
+            if move.l10n_in_state_id == self.env.ref('l10n_in.state_in_oc'):
+                country_id = self.env["res.country"].search([("code", "!=", "IN")], limit=1)
```

Comment:
Avoid additional queries in loops
```suggestion
            if move.l10n_in_state_id.l10n_in_tin == 96:  # foreign country
                country_id = other_country = other_country or self.env["res.country"].search([("code", "!=", "IN")], limit=1)
```

Or maybe like this if the version above is confusing
```suggestion
            if move.l10n_in_state_id.l10n_in_tin == 96:  # foreign country
                other_country = other_country or self.env["res.country"].search([("code", "!=", "IN")], limit=1)
                country_id = other_country
```